triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Server.scala

```diff
@@ -600,15 +600,28 @@ object Server {
     }
 
     def restartTrigger(req: RestartTrigger): Unit = {
-      val _ = spawnTrigger(req.trigger, req.runningTrigger, req.compiledPackages)
+      // If the trigger is still running we need to shut it down first
+      // and wait for it to terminate before we can spawn it again.
+      // Otherwise, akka will raise an error due to a non-unique actor name.
+      getRunnerRef(req.runningTrigger.triggerInstance) match {
```

Comment:
This feels a bit backwards. We could just watch when we start it and throw the exception when we failed. I think I proposed that before but it doesn’t work because we cannot throw from the behavior interceptor? If so, totally fine to keep it until the akka bug is fixed.

Reply:
We only want to restart after a token expiry and after we've refreshed the token. So we can't `watchWith(..., RestartTrigger(...))` when we start the trigger. 
We could let the server watch the `TriggerRunner` and then handle refresh and restart in the `ChildFailed` signal handler. As described in https://github.com/digital-asset/daml/pull/8037#discussion_r534048335 this is awkward right now as we don't have the necessary information available. I'll see if we can refactor the trigger server to make this easier.